### ISSUE TITLE
My problem has been solved because the configuration name of redis in version 2.3 is not quite the same as in the example

### ISSUE THREAD

**Issue by JWenBin** [RedisLabs/spark-redis#173]
My problem has been solved because the configuration name of redis in version 2.3 is not quite the same as in the example

_Originally posted by @liuxiaoji in https://github.com/RedisLabs/spark-redis/issues/172#issuecomment-497677221_

原来是我在2.4用了你2.3的配置参数；你在2.3用了我2.4的配置参数。。。。。o(╯□╰)o

**Comment by fe2s** [comment]
Could you please elaborate more on this?
Which example do you refer?

**Comment by liuxiaoji** [comment]
https://github.com/RedisLabs/spark-redis/blob/master/doc/getting-started.md
I can not find the 2.3 version of the example on the reference here, later found that there are problems to look at the source code

**Comment by fe2s** [comment]
You have to switch to the `branch-2.3`: https://github.com/RedisLabs/spark-redis/blob/branch-2.3/doc/getting-started.md

I still don't get it, is there something wrong with the example or configuration parameters?

**Comment by liuxiaoji** [comment]
The version I introduced with maven is 2.3.1. I just saw that the parameters in the 2.3 branch example are the same as those in the 2.4 version. I did not connect to redis after following the example, but I changed the code:
Val sc = new SparkContext(new SparkConf())
. SetAppName (" myApp ")
Initial redis host - can be any node in cluster mode
Set (" redis. The host, "PRODHOST)
/ / initial redis port
Set (" redis. Port ", "6379")
// optional redis AUTH password
Set (" redis. Auth, "PRODPASS)
)
With this code, I  can normally connect to redis

**Comment by liuxiaoji** [comment]
you can see sc.fromRedisKV[T]()-->new RedisConfig(new RedisEndpoint(sc.getConf))-->new RedisEndpoint(sc.getConf)--> def this(conf: SparkConf) {
      this(
        conf.get("redis.host", Protocol.DEFAULT_HOST),
        conf.getInt("redis.port", Protocol.DEFAULT_PORT),
        conf.get("redis.auth", null),
        conf.getInt("redis.db", Protocol.DEFAULT_DATABASE),
        conf.getInt("redis.timeout", Protocol.DEFAULT_TIMEOUT)
      )
  }
I changed the configuration after reading the above code

**Comment by fe2s** [comment]
The config names were changed in Spark-Redis 2.3.1 by adding 'spark.xxx' prefix to fix issue #93 

In 2.3.0
https://github.com/RedisLabs/spark-redis/blob/v2.3.0/src/main/scala/com/redislabs/provider/redis/RedisConfig.scala
```
conf.get("redis.host", Protocol.DEFAULT_HOST),
conf.getInt("redis.port", Protocol.DEFAULT_PORT),
conf.get("redis.auth", null),
conf.getInt("redis.db", Protocol.DEFAULT_DATABASE),
conf.getInt("redis.timeout", Protocol.DEFAULT_TIMEOUT)
```
 
In 2.3.1
https://github.com/RedisLabs/spark-redis/blob/v2.3.1/src/main/scala/com/redislabs/provider/redis/RedisConfig.scala
```
conf.get("spark.redis.host", Protocol.DEFAULT_HOST),
conf.getInt("spark.redis.port", Protocol.DEFAULT_PORT),
conf.get("spark.redis.auth", null),
conf.getInt("spark.redis.db", Protocol.DEFAULT_DATABASE),
conf.getInt("spark.redis.timeout", Protocol.DEFAULT_TIMEOUT)
```

The documentation for the `branch-2.3` is here https://github.com/RedisLabs/spark-redis/blob/branch-2.3/doc/getting-started.md and it mentions the new config parameters, i.e. prefixed with 'spark'. 

So the confusion might be if someone tries to use older version 2.3.0 and finds newer param names. 
For the version 2.3.1 the documentation is correct. Was that your case?

To address that we should update the docs saying that for the version 2.3.0 or below the param names are different.

**Comment by liuxiaoji** [comment]
Then I see.Because the document I was looking at was not the version of the document I used, the error occurred.
You can update the document as you said, which is still good for users who use the old version.
